File: lib/models/character.dart
import 'package:flutter/foundation.dart';
class Character {
  final String id;
  final String name;
  final String backstory;
  final String? avatarUrl;
  final String? visualStyle;
  final List<Map<String, String>> skills;         
  final List<Map<String, String>> achievements; 
  final String visualDescription;

  Character({
    required this.id,
    required this.name,
    required this.backstory,
    this.avatarUrl,
    required this.visualStyle,
    required this.visualDescription,
    required this.skills,
    required this.achievements,
  });

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'backstory': backstory,
    'avatar_url': avatarUrl,
    'visual_style': visualStyle,
    'visual_description': visualDescription,
    'skills': skills,
    'achievements': achievements,
  };

  factory Character.fromJson(Map<String, dynamic> json) => Character(
    id: json['id'] as String, 
    name: json['name'] as String,
    backstory: json['backstory'] as String,
    avatarUrl: json['avatar_url'] as String?,
    visualStyle: json['visual_style'] as String?,
    visualDescription: json['visual_description'] as String? ?? '',
    skills: (json['skills'] as List?)?.map((e) => Map<String, String>.from(e)).toList() ?? [],
    achievements: (json['achievements'] as List?)?.map((e) => Map<String, String>.from(e)).toList() ?? [],
  );
}

File: backend/services/postgre_service/character.py
import asyncpg
from postgre_conn import get_pool
from core.schemas import CharacterSchema


async def create_character(character: CharacterSchema) -> None: 
    """Создает персонажа"""
    pool=get_pool()
    async with pool.acquire() as conn: 
      row=conn.fetchrow('''
    INSERT INTO character (name, backstory, avatar_url, visual_style, visual_description)
    VALUES ($1, $2, $3, $4, $5)
    RETURNING id
    ''',
    character.name,
    character.backstory,
    character.avatar_url,
    character.visual_style,
    character.visual_description,
        )
    character_id = row["id"]   

    # Сохраняем навыки (если есть)
    for skill in character.skills:
        await conn.execute("""
            INSERT INTO character_skill (character_id, name, description)
            VALUES ($1, $2, $3)
            ON CONFLICT (character_id, name)
            DO UPDATE SET description = EXCLUDED.description
            """,
            character_id,
            skill["name"],
            skill.get("description"),
            )
    for achievement in character.achievements:
            await conn.execute(
                """
                INSERT INTO character_achievement (character_id, name, description)
                VALUES ($1, $2, $3)
                ON CONFLICT (character_id, name)
                DO UPDATE SET description = EXCLUDED.description
                """,
                character_id,
                achievement["name"],
                achievement.get("description"),
            )
    return character_id